### ISSUE TITLE
Unrecognized Error while doing transfer inside receive fallback method

### ISSUE THREAD

**Issue by crisgarner** [nomiclabs/buidler#791]
I have one orchestrator contract which handle actions for another contract. I'm getting this error:

`     Error: Transaction reverted for an unrecognized reason. Please report this to help us improve Buidler.
`

The flow is the following, call retrieveFees on orchestrator: 
``` solidity
//orchestrator.sol
function retrieveFees(address _child)
    public
    onlyOwner
{
    _child.retrieveFees();
}

receive() external payable {
    uint256 amount = address(this).balance;
    console.log(amount);
    payable(owner()).transfer(amount);
}
```

``` solidity
///child.sol

function retrieveFees() public virtual onlyOwner {
    //owner is the orchestrator
    uint256 amount = address(this).balance;
    payable(owner()).transfer(amount);
    emit LogRetrieveFees(msg.sender, amount);
}

```

**Comment by crisgarner** [comment]
This, seem to be an out of gas error.

**Comment by fvictorio** [comment]
Hi @crisgarner, do you have a public repo where we can debug this? Also, when does this happen? Running a test?

**Comment by crisgarner** [comment]
> Hi @crisgarner, do you have a public repo where we can debug this? Also, when does this happen? Running a test?

Hey, so I changed the code in order to make it work.  on a test, I was calling retrieveFees from the orchestrator. Which class the retrieveFees on the child, which sends a transfer to the orchestrator which triggers the receive() external function. To fix it I just made this a two transaction instead of just one. That's the reason my guess is an out of gas error.
